client/landing/jetpack-cloud/sections/settings/main.jsx

```diff
@@ -27,55 +29,104 @@ import './style.scss';
+	if ( isConnected ) {
+		return connectedProps( translate, message );
+	}
+	return disconnectedProps( translate, message );
+};
 
-	return (
-		<Card compact={ true } className={ cardProps.className }>
-			<img className="settings__icon" src={ cardProps.iconPath } alt="" />
-			<div className="settings__details">
-				<div className="settings__details-head"> { cardProps.title } </div>
-				<div>{ cardProps.content }</div>
-			</div>
-		</Card>
-	);
+const ConnectionStatus = ( { cardProps } ) => (
+	<Card compact={ true } className={ cardProps.className }>
+		<img className="settings__icon" src={ cardProps.iconPath } alt="" />
+		<div className="settings__details">
+			<div className="settings__details-head"> { cardProps.title } </div>
+			<div>{ cardProps.content }</div>
+		</div>
+	</Card>
+);
+
+const getStatusMessage = ( isConnected, hasBackup, hasScan, translate ) => {
+	const HAS_BACKUP = 1;
+	const HAS_SCAN = 2;
+	const HAVE_BOTH = 3;
+
+	const helpLink = {
+		components: {
+			a: (
+				<ExternalLink
+					className="settings__link-external"
+					icon
+					href="https://jetpack.com/support/adding-credentials-to-jetpack/"
+				/>
+			),
+		},
+	};
+
+	const disconnectedMessages = {
+		[ HAS_BACKUP ]: translate(
+			'Enter your server credentials to enable one-click restores for backups. {{a}}Need help? Find your server credentials{{/a}}',
+			helpLink
+		),
+		[ HAS_SCAN ]: translate(
+			'Enter your server credentials to enable Jetpack to auto-fix threats. {{a}}Need help? Find your server credentials{{/a}}',
+			helpLink
+		),
+		[ HAVE_BOTH ]: translate(
+			'Enter your server credentials to enable one-click restores for backups and to auto-fix threats. {{a}}Need help? Find your server credentials{{/a}}',
+			helpLink
+		),
+	};
+
+	const connectedMessages = {
+		[ HAS_BACKUP ]: translate( 'One-click restores are enabled.' ),
+		[ HAS_SCAN ]: translate( 'Auto-fix threats are enabled.' ),
+		[ HAVE_BOTH ]: translate( 'One-click restores and auto-fix threats are enabled.' ),
+	};
+
+	const messages = isConnected ? connectedMessages : disconnectedMessages;
+
+	if ( hasBackup && hasScan ) {
+		return messages[ HAVE_BOTH ];
+	} else if ( hasBackup ) {
+		return messages[ HAS_BACKUP ];
+	} else if ( hasScan ) {
+		return messages[ HAS_SCAN ];
+	}
+
+	return '';
 };
 
 const SettingsPage = () => {
 	const translate = useTranslate();
 	const siteId = useSelector( getSelectedSiteId );
 
-	const rewind = useSelector( ( state ) => getRewindState( state, siteId ) );
-	const isInitialized = rewind.state !== 'uninitialized';
-	const isConnected = rewind.state === 'active';
+	const scanState = useSelector( ( state ) => getSiteScanState( state, siteId ) );
+	const backupState = useSelector( ( state ) => getRewindState( state, siteId ) );
+
+	const isInitialized = backupState.state !== 'uninitialized';
+	const isConnected = backupState.state === 'active';
+
+	const hasBackup = backupState?.state !== 'unavailable';
```

Comment:
As the code is written currently, is it possible to get a state where the user is connected but does not have Backup?

The way I'm reading it, `isConnected` becomes false if `backupState` is anything but "active," so in our connection status message, `hasBackup = false` and `isConnected = true` wouldn't be allowed.

Possible fix: maybe `isConnected` should be an OR of `scanState` and `backupState`?

Reply:
This may also apply for `isInitialized`, which also looks only at `backupState` right now.

Reply:
And in the reverse case, `hasBackup`/`hasScan` could also be affected.

Reply:
Not sure about it. This needs a rethink since "need_credentials" only works for Rewind. The Scan API doesn't check it. Does the Scan API need server access to work?

Here https://opengrok.a8c.com/source/xref/trunk/wp-content/lib/activity-log/rewind-credentials.php#47 it seems like we need VaultPress system to do it, so I don't sure how deal with that, having Scan product and not a Backup product.